### PR TITLE
8341978: Improve JButton/bug4490179.java

### PR DESCRIPTION
**Pull request by aivanov-jdk**
At first, I noticed bug4490179.java didn't release mouse button 1, which could affect the following tests.

I started looking at the test and updated it to use `CountDownLatch` instead of delays.

@azvegint handled releasing mouse button 1 in https://github.com/openjdk/jdk/pull/19339#discussion_r1609201116.

The updated version first verifies `mouseButton3Released` event is received and ensures `actionPerformed` is not received at this time.

Then mouse button is released; this should trigger two events: `mouseButton1Released` and `actionPerformed`. It is verified that these events are received.

I kept debugging prints, they help understanding which actions lead to which events.

The updated test reproduces the original, it is easily verified using Java 8; the updated test is stable in CI.

The `javax/swing/JButton/bug4490179.java` test is also modified in https://github.com/openjdk/jdk/pull/20861, @mrserb, could you take a look? I was going to suggested factoring out the changes to this test from #20861, yet I modified it myself. I realised it when I started creating the PR.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341978](https://bugs.openjdk.org/browse/JDK-8341978): Improve JButton/bug4490179.java (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * @abdelhak-zaaim (no known openjdk.org user name / role)
 * [Alisen Chung](https://openjdk.org/census#achung) (@alisenchung - Committer)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21474/head:pull/21474` \
`$ git checkout pull/21474`

Update a local copy of the PR: \
`$ git checkout pull/21474` \
`$ git pull https://git.openjdk.org/jdk.git pull/21474/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21474`

View PR using the GUI difftool: \
`$ git pr show -t 21474`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21474.diff">https://git.openjdk.org/jdk/pull/21474.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21474#issuecomment-2407805254)

### CODE REVIEW

**Comment by aivanov-jdk** [comment]
/summary
Use latch to synchronize actions in the test.
Verify mouseButton3Released does not trigger actionPerformed
while mouse button 1 is pressed.
Ensure mouse button 1 is released.
Verify releasing mouse button 1 triggers actionPerformed.

**Review by azvegint — APPROVED** [review]

**Review by abdelhak-zaaim — APPROVED** [review]

File: test/jdk/javax/swing/JButton/bug4490179.java [hunk 1]
```diff
@@ -31,61 +31,107 @@
...
     static JFrame frame;
     static JButton button;
-    static volatile Point pt;
-    static volatile int buttonW;
-    static volatile int buttonH;
-    static volatile boolean passed = true;
+
+    private static volatile Point buttonCenter;
+
+    private static final CountDownLatch windowGainedFocus = new CountDownLatch(1);
```
Comment (alisenchung) [inline test/jdk/javax/swing/JButton/bug4490179.java:58]: line here is too long and probably should be put in 2 lines

File: test/jdk/javax/swing/JButton/bug4490179.java [hunk 2]
```diff
@@ -31,61 +31,107 @@
...
-                throw new RuntimeException("Test Failed");
+            try {
+                if (!mouseButton3Released.await(1, SECONDS)) {
+                    throw new RuntimeException("Mouse button 3 isn't released");
+                }
+
+                robot.waitForIdle();
+
+                if (actionPerformed.await(100, MILLISECONDS)) {
+                    throw new RuntimeException("Action event triggered by releasing button 3");
```
Comment (alisenchung) [inline test/jdk/javax/swing/JButton/bug4490179.java:123]: line here also should be put into 2 lines

**Review by alisenchung — APPROVED** [review]
just some formatting comments, test is working fine for me on macos

**Review by mrserb — APPROVED** [review]

File: test/jdk/javax/swing/JButton/bug4490179.java (hunk 2 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/JButton/bug4490179.java:123]: I'm reluctant about changing it… The PR is approved now. And the line still fits 100-column limit.

Yet here it wrapping the line has its value, the button number is at the end of the line.

Comment (aivanov-jdk) [reply test/jdk/javax/swing/JButton/bug4490179.java:123]: I talked to @azvegint and we decided it's good enough. Wrapping the lines would require re-review.

**Pull request closed without merging**